src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -10568,10 +10638,10 @@ namespace WaterThermalTanks {
         WaterThermalTank(WaterThermalTankNum).SourceMassFlowRate = Par(5) * HPPartLoadRatio;
         // FirstHVACIteration is a logical, Par is real, so make 1.0=TRUE and 0.0=FALSE
         FirstHVACIteration = (Par(4) == 1.0);
-        CalcWaterThermalTankMixed(WaterThermalTankNum);
+        CalcWaterThermalTank(WaterThermalTankNum);
         NewTankTemp = WaterThermalTank(WaterThermalTankNum).TankTemp;
-        PLRResidualMixedTank = Par(1) - NewTankTemp;
-        return PLRResidualMixedTank;
+        PLRResidualWaterThermalTank = Par(1) - NewTankTemp;
+        return PLRResidualWaterThermalTank;
```

Comment:
Makes sense to me. Did we determine that you didn't need to update the coil here? I remember at one point you said you were hitting iteration limits in your .err file. Maybe I'm just imagining that.

Reply:
Oh yeah I did come to some iteration warnings when I checked error file which might need to explore more. I decide to not update coil PLR and QHeaterRate  since the parameters I used from desuperheater coil is only for Qheatrate calculation(it is not directly stored in the struct so I used those two, it is the heat available got from HVAC part, and it is not changed at each system time step, QHeatRate = QHeaterRate/PLR), the solveroot function is to update PLR then change mass flow rate to reflect the heater's capacity change.